### PR TITLE
OpenAI Feed: Handle sale_price_effective_date even when there is no defined date range

### PR DESCRIPTION
**Pull request by htdat**
Fixes WOOPTP-49

## Description

Fixes handling of `sale_price_effective_date` when sale dates are partially or completely missing.

1. **Schema and Validatioion update** (33aa606): 
   - Removed `required_when` and `date_range` type constraints for `sale_price_effective_date` field. 
   - Update `validate_sale_dates` to strictly follow the specs validation. 
2. **ProductMapper enhancement** (24551f6): Updated `get_sale_date_range()` to generate default dates when missing:
   - If `sale_from` is missing → defaults to today
   - If `sale_to` is missing → defaults to today + 1 month
   - Hook `wpfoai_sale_price_default_start_date` - Override default start date
   - Hook `wpfoai_sale_price_default_end_date` - Override default end date
   - Hook `wpfoai_sale_price_effective_date` - Override complete date range

3. **Add unit tests** for all changes. 

## Testing instructions

1. Create a product with sale price but **no sale dates**:
   - Verify `sale_price_effective_date` is generated with today and today+1month
2. Create a product with sale price and **only start date**:
   - Verify end date defaults to today+1month
3. Create a product with sale price and **only end date**:
   - Verify start date defaults to today
4. Create a product with **both sale dates**:
   - Verify dates are used as-is (existing behavior)
6. Create a product **without sale price**:
   - Verify `sale_price_effective_date` is not present in feed

## Checklist

- [x] `npm run test:php` does not return errors.
- [x] `npm run lint:php` does not return errors.

## Summary by CodeRabbit

* **Bug Fixes**
  * sale_price_effective_date now must be provided when a sale price exists and must match "YYYY-MM-DD / YYYY-MM-DD".
  * When sale_price is present but dates are missing, start/end default to today and today+30 days, producing a valid range string.

* **Tests**
  * Added comprehensive unit tests covering valid/invalid formats, required-with-sale behavior, defaulting, and edge cases.

### CODE REVIEW

File: src/Integrations/OpenAi/ProductMapper.php [hunk 1]
```diff
@@ -879,11 +883,48 @@ private function get_sale_date_range( \WC_Product $product ): ?string {
...
+		 * Filter the complete sale price effective date range.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param string       $date_range Formatted date range (YYYY-MM-DD / YYYY-MM-DD).
+		 * @param \WC_Product  $product    Product object.
+		 * @param \WC_DateTime $sale_from  Start date object.
+		 * @param \WC_DateTime $sale_to    End date object.
+		 */
+		return apply_filters( 'wpfoai_sale_price_effective_date', $date_range, $product, $sale_from, $sale_to );
```
Comment (RadoslavGeorgiev) [inline src/Integrations/OpenAi/ProductMapper.php]: We have the `wpfoai_map_product` top-level filter. It feels like overkill to add 3 further filters for a single field, considering we are expecting to go through hundreds of thousands of products.

**Review by RadoslavGeorgiev — APPROVED** [review]
Look good overall, with a couple of nitpicks:

1. What the rabbit said.
2. The remark regarding filters. I think we can minimize them significantly.

Merge once both are addressed.

File: src/Integrations/OpenAi/ProductMapper.php (hunk 1 above)
Comment (htdat) [reply src/Integrations/OpenAi/ProductMapper.php]: Make sense, removed them 6bf1b4f1e3f25aab349bdb2f7e7a2286df878de0

File: src/Integrations/OpenAi/ProductMapper.php [hunk 2]
```diff
@@ -879,11 +883,48 @@ private function get_sale_date_range( \WC_Product $product ): ?string {
...
+			 *
+			 * @since 0.1.0
+			 *
+			 * @param \WC_DateTime $sale_to Default end date (today + 1 month).
+			 * @param \WC_Product  $product Product object.
+			 */
+			$sale_to = apply_filters( 'wpfoai_sale_price_default_end_date', $sale_to, $product );
+		}
+
+		$date_range = $sale_from->date_i18n( 'Y-m-d' ) . ' / ' . $sale_to->date_i18n( 'Y-m-d' );
```
[the remark this answers is not in the document]
Comment (htdat) [reply src/Integrations/OpenAi/ProductMapper.php]: Fixed here 69616fbe6cdbddbe484daac04e5fb276eb217161, which accounts for the far past date, such as sale start date `3 months ago`. Then, it ensures that it will pick today + 30 days.
